src/codecs/tiff.rs

```diff
@@ -352,24 +433,53 @@ fn cmyk_to_rgb(cmyk: &[u8]) -> [u8; 3] {
     ]
 }
 
-// Utility to simplify and deduplicate error handling during 16-bit encoding.
-fn u8_slice_as_u16(buf: &[u8]) -> ImageResult<&[u16]> {
-    bytemuck::try_cast_slice(buf).map_err(|err| {
-        // If the buffer is not aligned or the correct length for a u16 slice, err.
-        //
-        // `bytemuck::PodCastError` of bytemuck-1.2.0 does not implement
-        // `Error` and `Display` trait.
-        // See <https://github.com/Lokathor/bytemuck/issues/22>.
-        ImageError::Parameter(ParameterError::from_kind(ParameterErrorKind::Generic(
-            format!("{err:?}"),
-        )))
-    })
+/// Convert a slice of sample bytes to its semantic type, being a `Pod`.
+fn u8_slice_as_pod<P: bytemuck::Pod>(buf: &[u8]) -> ImageResult<std::borrow::Cow<'_, [P]>> {
+    bytemuck::try_cast_slice(buf)
+        .map(std::borrow::Cow::Borrowed)
+        .or_else(|err| {
+            match err {
+                bytemuck::PodCastError::TargetAlignmentGreaterAndInputNotAligned => {
+                    // If the buffer is not aligned for a native slice, copy the buffer into a Vec,
+                    // aligning it in the process. This is only done if the element count can be
+                    // represented exactly.
+                    let vec = bytemuck::allocation::pod_collect_to_vec(buf);
+                    Ok(std::borrow::Cow::Owned(vec))
+                }
+                /* only expecting: bytemuck::PodCastError::OutputSliceWouldHaveSlop */
+                _ => {
+                    // `bytemuck::PodCastError` of bytemuck-1.2.0 does not implement `Error` and
+                    // `Display` trait.
+                    // See <https://github.com/Lokathor/bytemuck/issues/22>.
+                    Err(ImageError::Parameter(ParameterError::from_kind(
+                        ParameterErrorKind::Generic(format!(
+                            "Casting samples to their representation failed: {err:?}",
+                        )),
+                    )))
+                }
+            }
+        })
 }
 
 impl<W: Write + Seek> TiffEncoder<W> {
     /// Create a new encoder that writes its output to `w`
     pub fn new(w: W) -> TiffEncoder<W> {
-        TiffEncoder { w }
+        let comp = CompressionType::default().into_tiff();
+        TiffEncoder { w, comp }
+    }
+
+    /// Create a new encoder that writes its output to `w` with `CompressionType` `compression`.
+    ///
+    /// It is best to view the options as a _hint_ to the implementation on the smallest or fastest
+    /// option for encoding a particular image. That is, using options that map directly to a TIFF
+    /// image parameter will use this parameter where possible. But variants that have no direct
+    /// mapping may be interpreted differently in minor versions. The exact output is expressly
+    /// __not__ part of the SemVer stability guarantee.
```

Comment:
This seems like a more natural note to put on `CompressionType::Deflate` to explain that the compression levels don't have well defined meaning. Getting the requested compression algorithm seems like something that should be guaranteed.